staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -449,33 +454,46 @@ func (c *AvailableConditionController) deleteAPIService(obj interface{}) {
 		}
 	}
 	klog.V(4).Infof("Deleting %q", castObj.Name)
-	c.enqueue(castObj)
+	c.enqueueService(castObj)
 }
 
 // there aren't very many apiservices, just check them all.
-func (c *AvailableConditionController) getAPIServicesFor(obj runtime.Object) []*apiregistration.APIService {
+func (c *AvailableConditionController) getAPIServicesFor(obj runtime.Object) []string {
```

Comment:
the entire implementation of getAPIServicesFor() should look something like this:

```
metadata, err := meta.Accessor(obj)
if err != nil {
  utilruntime.HandleError(err)
  return nil
}
c.RLock()
defer c.RUnlock()
return c.cache[metadata.GetNamespace()][metadata.GetName()]
```

the work to build/maintain c.cache would be done in the event handlers for APIService